fb-negotiation/sessions/rulebased_session.py

```diff
@@ -144,21 +134,90 @@ def propose(self):
+            proposal_string = str(self.my_proposal[self.top_item])
+        elif item == 'mid':
+            proposal_string = str(self.my_proposal[self.middle_item])
+        elif item == 'btm':
+            proposal_string = str(self.my_proposal[self.bottom_item])
+        return proposal_string + " "
+
+    def init_propose(self):
+        self.my_proposal[self.top_item] = 1
+        self.my_proposal[self.middle_item] = 0
+        self.my_proposal[self.bottom_item] = 0
+        self.my_proposal['made'] = True
+
+        if self.strategy == 'overvalued':
+            test_proposal = copy.deepcopy(self.my_proposal)
+            test_proposal[self.middle_item] += 1
+            if self.meets_bottomline(test_proposal):
+                self.my_proposal[self.middle_item] += 1
+            else:
+                self.my_proposal[self.middle_item] += 2
+
+        elif self.strategy == 'balanced':
+            test_proposal = copy.deepcopy(self.my_proposal)
+            test_proposal[self.middle_item] += 1
+            if self.meets_bottomline(test_proposal):
+                self.my_proposal[self.middle_item] += 1
+            else:
+                test_proposal[self.bottom_item] += 1
+                if self.meets_bottomline(test_proposal):
+                    self.my_proposal[self.middle_item] += 1
+                    self.my_proposal[self.bottom_item] += 1
+                else:
+                    test_proposal[self.top_item] += 1
+                    if self.valid_proposal(test_proposal)[0]:
+                        self.my_proposal[self.top_item] += 1
+                        self.my_proposal[self.middle_item] += 1
+                        self.my_proposal[self.bottom_item] += 1
+                    else:
+                        self.my_proposal[self.middle_item] += 2
+                        self.my_proposal[self.bottom_item] += 1
+
+        s = "How about " + self.p_str('top') + str(self.top_item) + ", " \
+                         + self.p_str('mid') + str(self.middle_item) + " and " \
+                         + self.p_str('btm') + str(self.bottom_item) + "."
+        self.state['last_act'] = 'propose'
+        return s
 
-    def agree(self):
-        self.state['last_act'] = 'agree'
+    def select(self):
+        self.state['last_act'] = 'select'
         # if random.random() < 0.5:
         #     return self.offer(self.my_price)
         # else:
         self.my_proposal['made'] = self.their_offer['made']
-        self.my_proposal['book'] = self.item_counts[0] - self.their_offer['book']
-        self.my_proposal['hat'] = self.item_counts[1] - self.their_offer['hat']
-        self.my_proposal['ball'] = self.item_counts[2] - self.their_offer['ball']
+        self.my_proposal['book'] = self.item_counts['book'] - self.their_offer['book']
+        self.my_proposal['hat'] = self.item_counts['hat'] - self.their_offer['hat']
+        self.my_proposal['ball'] = self.item_counts['ball'] - self.their_offer['ball']
 
```

Comment:
Let's use a for loop for these statements so that it's easy to extend to more items. What do you think?